### PR TITLE
vulkan: 1.1.77.0 -> 1.1.82.1

### PR DESCRIPTION
**Pull request by Ralith**
###### Motivation for this change
Lockstep update of all vulkan packages.

Subsumes #45124 and #45125.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/compilers/glslang/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation rec {
   name = "glslang-git-${version}";
-  version = "2018-06-21";
+  version = "2018-08-16";
 
   src = fetchFromGitHub {
-    owner = "KhronosGroup";
+    owner = "lenny-lunarg";
```
Comment (ajs124) [inline pkgs/development/compilers/glslang/default.nix]: Why change the source to this fork?

Comment (Ralith) [reply pkgs/development/compilers/glslang/default.nix]: Upstream forked to backport an important bugfix. See https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/252 for details. The source will be reverted in the next update.

**Comment by Ralith** [comment]
Fixed refs. Annoyingly, upstream didn't tag 1.1.82.1 anywhere, though since it's only a glslang bump all the vulkan-* source is actually unchanged.

File: pkgs/development/libraries/vulkan-headers/default.nix [hunk 2]
```diff
@@ -1,15 +1,15 @@
 { stdenv, fetchFromGitHub, cmake }:
 stdenv.mkDerivation rec {
   name = "vulkan-headers-${version}";
-  version = "1.1.77.0";
+  version = "1.1.82.1";
```
[the remark this answers is not in the document]
Comment (Ralith) [reply pkgs/development/libraries/vulkan-headers/default.nix]: I could restrict the version change to -validation-layers; would that be preferable?

File: pkgs/development/compilers/glslang/default.nix (hunk 1 above)
Comment (Ralith) [reply pkgs/development/compilers/glslang/default.nix]: Looks that way. Are you thinking incorporating the patch nix-side would be better?

Comment (Mic92) [reply pkgs/development/compilers/glslang/default.nix]: You can use the upstream url and `fetchpatch` then.

File: pkgs/development/libraries/vulkan-headers/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/development/libraries/vulkan-headers/default.nix]: Keeping `version` and `rev` in sync looks less confusing to me.

**Comment by Ralith** [comment]
Reverted to more conservative versioning and switched to fetchpatch for the glslang fix.

**Comment by Ralith** [comment]
Any hope of getting this in time for the branch?

**Pull request merged**